src/fromager/resolver.py

```diff
@@ -188,11 +189,13 @@ def __init__(
         include_sdists: bool = True,
         include_wheels: bool = True,
         sdist_server_url: str = "https://pypi.org/simple/",
+        constraints: Constraints | None = None,
     ):
         super().__init__()
         self.include_sdists = include_sdists
         self.include_wheels = include_wheels
         self.sdist_server_url = sdist_server_url
+        self.constraints = constraints if constraints else Constraints({})
```

Comment:
You don't have to change it, but you can simplify this to 

```suggestion
        self.constraints = constraints or Constraints({})
```